back/app/controllers/web_api/v1/projects_allowed_input_topics_controller.rb

```diff
@@ -0,0 +1,59 @@
+class WebApi::V1::ProjectsAllowedInputTopicsController < ApplicationController
+  before_action :set_projects_allowed_input_topic, only: %i[show reorder destroy]
+  skip_before_action :authenticate_user
+
+  def index
+    @projects_allowed_input_topics = policy_scope(ProjectsAllowedInputTopic)
+    @projects_allowed_input_topics = @projects_allowed_input_topics.where(project_id: params[:project_id]) if params[:project_id].present?
```

Comment:
You're right, for the index action it doesn't make sense. I have edited the routes to be:
```
resources :projects_allowed_input_topics, only: [:show, :create, :reorder, :destroy] do
        patch 'reorder', on: :member
        get '/project/:project_id/allowed_input_topics', on: :collection, to: 'projects_allowed_input_topics#index'
end
```
  
Note that I also removed `:index` from the `only: [ ... ]` list. Is this correct?

Reply:
I think you are right, it does not make sense to list all `allowed_input_topics` without also limiting the list to those related to a specific project. I have added a nested route (and removed `:index` from the `resources ... only: [ ... ]` list), to give:
```
resources :projects_allowed_input_topics, only: [:show, :create, :reorder, :destroy] do
        patch 'reorder', on: :member
        get '/project/:project_id/allowed_input_topics', on: :collection, to: 'projects_allowed_input_topics#index'
end
```
Is this what you advised?

Reply:
@sebastienhoorens Using nested routes, i was only able to reduced the non-nested routes to this:
```
resources :projects_allowed_input_topics, only: [:index, :show, :create, :destroy] do
    patch 'reorder', on: :member
end
```
`:index` is only required here to enable the request tested in the spec `get 'web_api/v1/projects_allowed_input_topics' in `back/spec/acceptance/projects_allowed_input_topics_spec.rb`, though I note I cannot see this endpoint in use in the FE (i.e. `front/app/services/projectTopics.ts` only uses this path for creation of a new item).  
  
`:create` is required here, probably because `front/app/services/projectTopics.ts` uses `${API_PATH}/projects_allowed_input_topics` for `addProjectTopic`